business/definitionService.js

```diff
@@ -190,6 +200,7 @@ class DefinitionService {
 
   async _store(definition) {
     await this.definitionStore.store(definition)
+    await this.cache.set(this._getCacheKey(definition.coordinates), definition)
```

Comment:
suggest putting the cache last. failing to put it in the cache would stop it going into the search which would be a functional problem (users could not find the definition). Failure to put it in the cache will just be a perf issue that should self-correct (if we add it when missing. see other comment above).